fvm/evm/emulator/state/code.go

```diff
@@ -0,0 +1,62 @@
+package state
+
+import (
+	"encoding/binary"
+	"fmt"
+)
+
+// CodeContainer contains codes and keeps
+// track of reference counts
+type CodeContainer struct {
+	code     []byte
+	refCount uint64
+}
```

Comment:
Maybe keep a reference to encoded buffer (from `CodeContainerFromEncoded()`) so we can reuse it during encoding.

```suggestion
type CodeContainer struct {
	code     []byte
	buffer    []byte
	refCount uint64
}
```

Reply:
good idea, updated to keep the encoded as buffer